deep_qa/layers/backend/squeeze.py

```diff
@@ -26,6 +26,11 @@ def compute_mask(self, inputs, mask=None):
         # pylint: disable=unused-argument
         if mask is None:
             return None
+        # If the mask doesn't need squeezing, just return it.
+        # TODO(becky): better to check if the last mask dim == 1?
+        if K.ndim(inputs) == K.ndim(mask) + 1:
```

Comment:
I'm pretty sure the shapes work out now that this isn't needed.  Can you remove it and see if it still works?